src/token.rs

```diff
@@ -5,37 +5,53 @@ pub enum TokenType {
     Plain,
     DoubleQuoted,
     SingleQuoted,
+    Variable,
 }
 
 #[derive(Clone, Debug, PartialEq)]
 pub enum Token {
     Meta(String),
     Plain(String),
-    DoubleQuoted(String),
-    SingleQuoted(String),
+    Variable(String),
+    DoubleQuoted(Vec<Token>),
+    SingleQuoted(Vec<Token>),
```

Comment:
I made this change because for this kind of input:

```
> echo "$FOO Hello world"
```

I needed a structure equivalent to (in actuality, "Hello world" is split into individual characters because it makes the code a lot easier):

```
DoubleQuoted([Variable("FOO"), "Hello world"])
```